packages/synapse-react-client/src/components/Forum/DiscussionThread.tsx

```diff
@@ -98,14 +100,36 @@ export function DiscussionThread(props: DiscussionThreadProps) {
     hasNextPage,
     fetchNextPage,
   } = useGetRepliesInfinite(threadId, orderByDatePosted, limit)
-  const replies = replyData?.pages.flatMap(page => page.results) ?? []
 
   const { data: moderatorList } = useGetModerators(threadData?.forumId ?? '')
 
   const isAuthorModerator = moderatorList?.results.includes(
     threadData?.createdBy ?? '',
   )
 
+  const params = new URLSearchParams(window.location.search)
+  const replyId = params.get('replyid')
+
+  useEffect(() => {
+    const allReplies = replyData?.pages.flatMap(page => page.results) ?? []
+    if (replyId) {
+      const filteredReplies = allReplies.filter(reply => reply.id === replyId)
+      setReplies(filteredReplies)
+    } else {
+      setReplies(allReplies)
+    }
+  }, [replyData?.pages, replyId])
+
+  const handleShowAllRepliesButton = () => {
+    const allReplies = replyData?.pages.flatMap(page => page.results) ?? []
+    setReplies(allReplies)
+
+    const params = new URLSearchParams(window.location.search)
+    params.delete('replyid')
+    const newUrl = `${window.location.pathname}?${params.toString()}`
+    window.history.replaceState({}, '', newUrl)
+  }
```

Comment:
Continued from above, make the URL param the single point of control over filtering logic. Just worried that the component doesn't reload and never catches the update to `window.location.search`, but hopefully it works.

Also suggest using `pushState` instead of `replaceState` so the user can go back to the reply-only if desired.

```suggestion
  const replies = useMemo(() => {
    const allReplies = replyData?.pages.flatMap(page => page.results) ?? []
    if (replyId) {
      return allReplies.filter(reply => reply.id === replyId)
    }
    return allReplies
  }, [replyId, replyData])

  const handleShowAllRepliesButton = () => {
    const params = new URLSearchParams(window.location.search)
    params.delete('replyid')
    const newUrl = `${window.location.pathname}?${params.toString()}`
    window.history.pushState({}, '', newUrl)
  }
```

Reply:
This doesn't seem to work for me. When the URL is modified the component doesn't re-render. Should window.location.search be in the useMemo hook?

Reply:
React won't know when `window.location.search` updates because it's not tracked internally. Can we add a `useNativeSearchParams` hook that is similar to the implementation here? https://www.30secondsofcode.org/react/s/use-hash-search-param/

Then use that to get `replyid`.